### DISCUSSION TITLE
[Feature] Search bar autocomplete

### DISCUSSION THREAD

**Discussion by ajnart** [immich-app/immich#5412] [Feature Request]
# Search bar autocomplete

## Idea
The search bar could be much better if it was made as some kind of spotlight like the current Github searchbar (press CTRL + K to bring it up) I would be really interested in having this kind of spotlight to be used for the following QOL improvements that are already in The-App-That-Shall-Not-Be-Named : 
- [ ] Autocomplete on albums
- [ ] Quickly jump to a year / month / date
- [ ] Autocomplete on faces (with a preview of the face)
- [ ] Autocomplete with the most common metadata (for example the Model for a camera if it is used often, the name of a camera lens) 
- And much more (Quickly move to settings, dark mode toggle instead of inside of the header, ...)

## Preview of the feature:
<img width="272" alt="image" src="https://github.com/immich-app/immich/assets/49837342/e5db0d90-0154-4a1b-8a24-fd426eb65799">

## Implementation
I would love to contribute, however I am not very familiar with this kind of monorepo, I saw [in the architecture page](https://immich.app/docs/developer/architecture) that there is a whole microservice for search but I don't feel like it's connected to the search bar before starting an actual search. If you could give me some pointers on where to start I would love to thinker around and see if I can implement this. 

### v1 : Basic autocomplete for Albums / People
When entering the search bar, a query would be done to get the name of all people / albums as well as their thumbnail, then I would make the search bar have an autocomplete feature that would show these items when trying as well as another entry like `Search for {querry}` 

### v2 : Spotlight / Action bar with navigation
In another version, this top bar could be improved to incorporate the features I talked about above.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Currently all the search stuff is a bit constrained because we're using Typesense. After we get rid of that (#3605), the goal is to give the search a major overhaul and make it actually useful with filters and such. This would definitely have a place in that, and a contribution is always welcome, but I would recommend waiting for that overhaul. 

If you want some specific guidance on developing for Immich or such, feel free to hop into the discord server so we can help you out!

**Reply by ajnart** [reply]
Alright thanks for the heads-up, I will try to run the dev environment on my computer and see if I can make some other quick changes I’ve thought about and I’ll come on the discord if I get stuck.

**Comment by bo0tzz** [comment]
#7264
